### PR TITLE
Ps/add strict optional mypy

### PR DESCRIPTION
**Pull request by Pastromhaug**
This makes DE mypy equally strict as scadmin.

### CODE REVIEW

File: simulator.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def get_next_unstarted_drill(self) -> Optional[str]:
         state = self.fetch_dialog_state(PHONE_NUMBER)
         assert state.user_profile
         language = state.user_profile.language
+        assert language
```
Comment (Pastromhaug) [inline simulator.py:69]: in the simulator I just added a bunch of asserts to make mypy happy

File: stopcovid/dialog/models/state.py [hunk 2]
```diff
@@ -45,15 +45,15 @@ def set_language(cls, value: Optional[str]) -> Optional[str]:
 class PromptState(pydantic.BaseModel):
     slug: str
     start_time: datetime.datetime
-    failures: Optional[int] = 0
-    reminder_triggered: Optional[bool] = False
+    failures: int = 0
+    reminder_triggered: bool = False
```
Comment (Pastromhaug) [inline stopcovid/dialog/models/state.py:49]: I looped through the entire `dialog-state-prod` table and didn't find any cases where these values were `None`

File: stopcovid/dialog/models/state.py [hunk 3]
```diff
@@ -45,15 +45,15 @@ def set_language(cls, value: Optional[str]) -> Optional[str]:
...
     last_response_time: Optional[datetime.datetime] = None
 
 
 class DialogState(pydantic.BaseModel):
     phone_number: str
     seq: str
-    user_profile: Optional[UserProfile] = pydantic.Field(
+    user_profile: UserProfile = pydantic.Field(
```
Comment (Pastromhaug) [inline stopcovid/dialog/models/state.py:56]: This is the biggest change in this whole PR. We assume the user profile is set everywhere and mypy is yelling about it. Looping through the `dialog-state-prod` table there appear to be 0 cases where `user_profile` is set to `None`

File: stopcovid/drills/drills.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Prompt(pydantic.BaseModel):
     messages: List[PromptMessage]
     response_user_profile_key: Optional[str] = None
     correct_response: Optional[str] = None
-    max_failures: Optional[int] = 1
+    max_failures: int = 1
```
Comment (Pastromhaug) [inline stopcovid/drills/drills.py:18]: I checked the`current_prompt` in the dialog state table, and verified we are defaulting to 1 when we send the prompts over from scadmin.

**Review by jeffreysilver — APPROVED** [review]

**Pull request merged**